src/main/java/build/buildfarm/instance/AbstractServerInstance.java

```diff
@@ -466,6 +474,14 @@ public String listOperations(
   }
 
   @Override
+  public void deleteOperation(String name) {
```

Comment:
If the operation is in completedOperations, it is removed.  If the operation is outstanding, an exception is thrown (changed this to IllegalStateException as you recommended).  If the operation is neither completed nor outstanding, the request does nothing.